src/librbd/ImageWatcher.cc

```diff
@@ -844,7 +844,6 @@ void ImageWatcher::handle_payload(const FlattenPayload &payload,
 
 void ImageWatcher::handle_payload(const ResizePayload &payload,
 				  bufferlist *out) {
-  RWLock::RLocker l(m_image_ctx.owner_lock);
```

Comment:
Could you explain a little bit more on why the cleanup logic doesn't function properly if we don't hold the lock here? Thanks!